bpfman-operator/apis/v1alpha1/cluster_tracepoint_program_types.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClTracepointProgramInfo contains the Tracepoint program details
+type ClTracepointProgramInfo struct {
+	// links is the list of points to which the program should be attached.  The list items
+	// are optional and may be updated after the bpf program has been loaded
+	// +optional
+	// +kubebuilder:default:={}
+	Links []ClTracepointAttachInfo `json:"links"`
+}
+
+type ClTracepointAttachInfo struct {
+	// name refers to the name of a kernel tracepoint to attach the
+	// bpf program to.
+	Name string `json:"name"`
```

Comment:
Any validations that can be put in place to ensure only valid kernel tracepoint names are specified?